File: solutions/cpp/Design HashSet.cpp
#include <iostream>
#include <vector>
class MyHashSet
{
public:
    std::vector<int> hashset;
    MyHashSet()
    {
        this->hashset = {};
    }

    void add(int key)
    {
        if (!contains(key))
        {
            hashset.push_back(key);
        }
    }

    void remove(int key)
    {
        if (contains(key))
        {
            int position = getPosition(key);
            hashset.erase(hashset.begin() + position);
        }
    }

    bool contains(int key)
    {
        int pos = getPosition(key);
        if (pos > -1)
        {
            return true;
        }
        else
        {
            return false;
        }
    }

    int getPosition(int key)
    {
        int i = 0, j = hashset.size() - 1;
        while (i <= j)
        {
            if (hashset[i] == key)
            {
                return i;
            }
            if (hashset[j] == key)
            {
                return j;
            }
            i = i + 1;
            j = j - 1;
        }
        return -1;
    }
};

File: solutions/Javascript/Design HashSet.js
var MyHashSet = function () {
  this.hashset = [];
};

MyHashSet.prototype.add = function (key) {
  if (this.hashset.indexOf(key) < 0) {
    this.hashset.push(key);
  }
};

MyHashSet.prototype.remove = function (key) {
  this.hashset = this.hashset.filter((num) => num !== key);
  return this.hashset;
};

MyHashSet.prototype.contains = function (key) {
  if (this.hashset.indexOf(key) > -1) {
    return true;
  } else {
    return false;
  }
};
